<launch>

    <arg name="markerId"          default="17"/>
    <arg name="markerSize"        default="0.07"/>     <!-- in m -->
	<arg name="camera_info"       default="/camera_info"/>
	<arg name="image_topic"       default="/image"/>
	<arg name="camera_frame"      default="/camera_frame"/>
    <arg name="marker_frame"      default="/aruco_marker_frame"/>
    <arg name="ref_frame"         default=""/>          <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" />    <!-- NONE, HARRIS, LINES, SUBPIX -->

    <node pkg="o2as_aruco_marker_detection" type="marker_detector" name="marker_detector" output="screen">
		<remap from="/camera_info"       to="$(arg camera_info)"/>
        <remap from="/image"             to="$(arg image_topic)" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>
        <param name="camera_frame"       value="$(arg camera_frame)"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

</launch>
